Cabal-QuickCheck/src/Test/QuickCheck/Instances/Cabal.hs

```diff
@@ -503,14 +503,25 @@ shortListOf1 bound gen = sized $ \n -> do
     vectorOf k gen
 
 arbitraryShortToken :: Gen String
-arbitraryShortToken = arbitraryShortStringWithout "{}[]"
+arbitraryShortToken = arbitraryShortStringWithout "{}[]" (const True)
 
 arbitraryShortPath :: Gen String
-arbitraryShortPath = arbitraryShortStringWithout "{}[],"
-
-arbitraryShortStringWithout :: String -> Gen String
-arbitraryShortStringWithout excludeChars =
-    shortListOf1 5 $ elements [c | c <- ['#' ..  '~' ], c `notElem` excludeChars ]
+arbitraryShortPath = arbitraryShortStringWithout "{}[],<>:|*?" (not . winDevice)
+    where
+        winDevice path = let p' = map toLower path
+                          in any (isDev p') ["con", "aux", "prn", "com", "lpt", "nul"]
+        isDev path dev = dev `isPrefixOf` path ||
+                         ('\\' : dev) `isInfixOf` path ||
+                         ('/' : dev) `isInfixOf` path
+
+arbitraryShortStringWithout :: String -> (String -> Bool) -> Gen String
+arbitraryShortStringWithout excludeChars validp =
+    shortListOf1 5 $ elements (str 5)
+    where
+        str :: Int -> String
+        str 0 = error "could not generate valid string"
+        str n = let s = [c | c <- ['#' ..  '~' ], c `notElem` excludeChars ]
```

Comment:
Suggestion: use list difference: `c <- ['#' .. '~' ] List.\\ excludeChars`

Reply:
That code was there to begin with; all I did was rearrange so I could exclude devices before turning it into a `Gen String`. I can still make the refactor.

Reply:
I refactored it locally, am still running validate before I push again.